README.md

```diff
@@ -0,0 +1,355 @@
+
+With this you can start simulating your process:
+
+- click on the play button on the starting event
+- click on the pause buttons found on all elements to stop the process at that point
+- on gateways you are able to choose which branch it should go to
+
+See the [plugin repository](https://github.com/camunda/camunda-modeler-token-simulation-plugin)
+for more information about it.
+
+#### Publish processes
+
+Once your process is ready for more real testing,
+the following steps are needed to get your process published:
+
+- the diagram must be saved in your file system
+- there should be no errors (see the bottom bar for the red circled error icon)
+    - the most common error is that the `StartEvent` element must define a `<History time to live>`
+        - for that click on the background canvas and select the `History cleanup` section on the right panel
+        - set the `time to live` field to a number, like `1`
+        - _if you don't see the right pane, go to the main menu -> `Window` -> `Toggle Properties Panel`_
+- [Camunda needs to be running](#camunda-platform-7) so you can push your processes there!
+
+Once these conditions are met, you are ready to click on the __rocket icon__
+located at the bottom bar.
+
+A small popup will show up to establish the connection to the Camunda server.
+
+The `REST endpoint` field should say `http://localhost:8081/engine-rest`,
+if it still shown an error message below the field,
+double check that your Camunda instance runs fine.
+
+On the `Authentication` section, select `HTTP Basic` and use `admin` for user and password.
+
+__Note:__ if your model includes forms, decision tables and/or other models
+you need to either upload them separately, or include them via the `Include additional files` section.
+
+Once everything is set up properly, finally, push the `Deploy` button!
+
+Head to the [Camunda Cockpit](#camunda-cockpit) to manage the processes.
+
+### Camunda Platform 7
+
+There is a [`docker-compose.yml`](./docker-compose.yml)
+with the main pieces to get Camunda working locally:
+
+- Camunda 7 itself
+- a PostgreSQL database server (where Camunda stores the tasks and processes)
+- a mail server (a mailhog instance, to check the mails being sent by processes)
+
+For that you will need to get `docker-compose` installed
+and a pair of [private keys generated](#authentication-keys).
+
+See the [`docker-compose` install instructions](https://docs.docker.com/compose/install/).
+
+To run Camunda, run:
+
+```shell
+docker-compose up
+```
+
+See the [troubleshooting sections if you are having problems](#troubleshooting).
+
+#### Authentication keys
+
+To get Plone and Camunda to talk to each other, they need to trust each other.
+
+For that a pair of public/private keys are used.
+
+__Plone__ needs a private key to communicate with Camunda via a JWT-authentication token,
+see the [installation instructions for Plone](#plone-integration),
+to generate it, run:
+
+```shell
```

Comment:
Let's discuss  later (after merg) what we shouuld put here for MacOSX users. the notes above are not a fix but only what I stumbled upon.

Reply:
I'm not sure if we need a specific cypher, or any random one is ok, as long as private and public key match 😓 

But I can add at least a comment for Mac OS X users meanwhile: `please switch to Linux 😝`

Reply:
nix run nixpkgs#openssl

AFAIK the current libraries (pyjwt in Python, bouncycastle+jose in Python) don’t have automatic cipher detection. Also, even JWT does include cipher info, keys must be parsed from PEM format first. So, I believe that we must choose the supported ciphers and manually support them in code. I picked recommended EC, because it should be fast, but big and slow RSA could be ubiquitously working fallback. Unless there are libraries that auto detect and support different algorithms by just giving any PEM file…